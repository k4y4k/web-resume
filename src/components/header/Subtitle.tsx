import * as React from "react";
import tw from "twin.macro";

interface SubtitleProps {
  subtitle: string;
  isCoverLetter?: boolean;
}

const Subtitle = ({ subtitle, isCoverLetter = false }: SubtitleProps) => {
  if (subtitle === "") return null;

  return (
    <h2
      css={[
        tw`my-2 text-2xl `,
        tw`text-orchid-600`,
        isCoverLetter && tw`text-white text-xl px-6`,
      ]}
      data-testid="headerSubtitle"
    >
      {subtitle}
    </h2>
  );
};

export default Subtitle;
