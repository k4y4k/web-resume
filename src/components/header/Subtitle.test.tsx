import { render, screen } from "@testing-library/react";
import * as React from "react";
import Subtitle from "./Subtitle";

describe("<Subtitle />", () => {
  test("Handles not having data", () => {
    render(<Subtitle lightmode={false} compact={false} subtitle="" />);
    const subtitle = screen.queryByTestId("headerSubtitle");

    expect(subtitle).not.toBeInTheDocument();
    expect(subtitle).toMatchSnapshot();
  });

  test("Displays subtitle", () => {
    render(
      <Subtitle
        lightmode={false}
        compact={false}
        subtitle="lead singer of kayak and the kayaks"
      />,
    );
    const subtitle = screen.getByTestId("headerSubtitle");

    expect(subtitle).toHaveTextContent("lead singer of kayak and the kayaks");
    expect(subtitle).toMatchSnapshot();
  });

  test("lightmode", () => {
    render(
      <Subtitle
        lightmode={true}
        compact={false}
        subtitle="lead singer of kayak and the kayaks"
      />,
    );
    const subtitle = screen.getByTestId("headerSubtitle");

    expect(subtitle).toHaveTextContent("lead singer of kayak and the kayaks");
    expect(subtitle).toMatchSnapshot();
  });

  test("compact", () => {
    render(
      <Subtitle
        lightmode={false}
        compact={true}
        subtitle="lead singer of kayak and the kayaks"
      />,
    );
    const subtitle = screen.getByTestId("headerSubtitle");

    expect(subtitle).toHaveTextContent("lead singer of kayak and the kayaks");
    expect(subtitle).toMatchSnapshot();
  });

  test("lightmode and compact", () => {
    render(
      <Subtitle
        lightmode={true}
        compact={true}
        subtitle="lead singer of kayak and the kayaks"
      />,
    );
    const subtitle = screen.getByTestId("headerSubtitle");

    expect(subtitle).toHaveTextContent("lead singer of kayak and the kayaks");
    expect(subtitle).toMatchSnapshot();
  });
});
