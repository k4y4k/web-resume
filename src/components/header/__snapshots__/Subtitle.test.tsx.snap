// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subtitle /> Displays subtitle 1`] = `
.c0 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  font-size: 1.5rem;
  line-height: 2rem;
  --tw-text-opacity: 1;
  color: rgba(103,73,177,var(--tw-text-opacity));
}

<h2
  class="c0"
  data-testid="headerSubtitle"
>
  lead singer of kayak and the kayaks
</h2>
`;

exports[`<Subtitle /> Handles not having data 1`] = `null`;

exports[`<Subtitle /> compact 1`] = `
.c0 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  font-size: 1.5rem;
  line-height: 2rem;
  --tw-text-opacity: 1;
  color: rgba(103,73,177,var(--tw-text-opacity));
}

<h2
  class="c0"
  data-testid="headerSubtitle"
>
  lead singer of kayak and the kayaks
</h2>
`;

exports[`<Subtitle /> lightmode 1`] = `
.c0 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  font-size: 1.5rem;
  line-height: 2rem;
  --tw-text-opacity: 1;
  color: rgba(103,73,177,var(--tw-text-opacity));
}

<h2
  class="c0"
  data-testid="headerSubtitle"
>
  lead singer of kayak and the kayaks
</h2>
`;

exports[`<Subtitle /> lightmode and compact 1`] = `
.c0 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  font-size: 1.5rem;
  line-height: 2rem;
  --tw-text-opacity: 1;
  color: rgba(103,73,177,var(--tw-text-opacity));
}

<h2
  class="c0"
  data-testid="headerSubtitle"
>
  lead singer of kayak and the kayaks
</h2>
`;
